src/cli/commands/workspace.js

```diff
@@ -39,8 +40,13 @@ export async function run(config: Config, reporter: Reporter, flags: Object, arg
     throw new MessageError(reporter.lang('workspaceUnknownWorkspace', workspaceName));
   }
 
+  const workspace = workspaces[workspaceName];
+
   try {
-    await child.spawn(process.argv[0], [process.argv[1], ...rest], {stdio: 'inherit'});
+    await child.spawn(NODE_BIN_PATH, [YARN_BIN_PATH, ...rest], {
```

Comment:
May be we should use [child_process.fork()](https://nodejs.org/api/child_process.html#child_process_child_process_fork_modulepath_args_options) for this instead?